From 95d4b62f40be95af05ce449c64a9a34e2da174a5 Mon Sep 17 00:00:00 2001
From: Adriana Reus <adriana.reus@nxp.com>
Date: Wed, 20 Sep 2017 10:58:01 +0300
Subject: [PATCH 2594/5242] MLK-16442-3: dts: imx8qm: Add clock parents for DC
 clocks.

commit  cec26d8776900aba903a4cb0b27d045e3d6e8475 from
https://source.codeaurora.org/external/imx/linux-imx.git

Adds dt settings for the dpu driver to set the default clock
parents
- PLL1 (dc0_pll0_clk) for dispay0 and PLL2(dc0_pll1_clk) for display1.
Functionality is not changed from dpu driver perspective as the same
parents for the display clocks were used before.

The resulting clock topology for dc0_disp1 is:

dc0_pll1_div                1            1  1188000000 0 0
    dc0_pll1_clk            2            2  1188000000 0 0
       dc0_disp1_sel        1            1  1188000000 0 0
          dc0_disp1_div     1            1   148500000 0 0
             dc0_disp1_clk  1            1   148500000 0 0

(BuildInfo: SCFW 9e9f6ec6, IMX-MKIMAGE 0, ATF 0)

Signed-off-by: Adriana Reus <adriana.reus@nxp.com>
Reviewed by: Ranjani Vaidyanathan <Ranjani.vaidyanathan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
index 9bc6d38..701ff27 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm.dtsi
@@ -1118,6 +1118,10 @@
 			 <&clk IMX8QM_DC0_DISP0_CLK>,
 			 <&clk IMX8QM_DC0_DISP1_CLK>;
 		clock-names = "pll0", "pll1", "disp0", "disp1";
+		assigned-clocks = <&clk IMX8QM_DC0_DISP0_SEL>,
+				  <&clk IMX8QM_DC0_DISP1_SEL>;
+		assigned-clock-parents = <&clk IMX8QM_DC0_PLL0_CLK>,
+					 <&clk IMX8QM_DC0_PLL1_CLK>;
 		power-domains = <&pd_dc0>;
 		status = "disabled";
 
@@ -1245,6 +1249,10 @@
 			 <&clk IMX8QM_DC1_DISP0_CLK>,
 			 <&clk IMX8QM_DC1_DISP1_CLK>;
 		clock-names = "pll0", "pll1", "disp0", "disp1";
+		assigned-clocks = <&clk IMX8QM_DC1_DISP0_SEL>,
+				  <&clk IMX8QM_DC1_DISP1_SEL>;
+		assigned-clock-parents = <&clk IMX8QM_DC1_PLL0_CLK>,
+					 <&clk IMX8QM_DC1_PLL1_CLK>;
 		power-domains = <&pd_dc1>;
 		status = "disabled";
 
-- 
1.7.9.5

